analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R

```diff
@@ -0,0 +1,174 @@
+    "results",
+    "controlfreec_annotated_cn_autosomes.tsv.gz"
+  )
+)
+
+# Read in fusion data
+fusion_df <- readr::read_tsv(
+  file.path(root_dir, "data", "pbta-fusion-putative-oncogenic.tsv"))
+
+# Read in output file from `01-HGG-molecular-subtyping-defining-lesions.Rmd`
+hgg_lesions_df <- readr::read_tsv(
+  file.path(
+    root_dir,
+    "analyses",
+    "molecular-subtyping-HGG",
+    "results",
+    "HGG_defining_lesions.tsv"
+  )
+)
+
+#### Filter HGG defining lesions data.frame ------------------------------------
+
+# Filter the output file from `01-HGG-molecular-subtyping-defining-lesions.Rmd`
+# for samples classified and reclassified as HGG
+hgg_lesions_df <- hgg_lesions_df %>%
+  dplyr::filter(
+    disease_type_new == "High-grade glioma;astrocytoma (WHO grade III/IV)" |
+      disease_type_reclassified == "High-grade glioma"
+  )
+
+#### Filter metadata -----------------------------------------------------------
+
+# Filter metadata for `High-grade glioma` and samples that should be classified
+# as High-grade glioma based on defining lesions
+hgg_metadata_df <- metadata %>%
+  dplyr::filter(
+    disease_type_new == "High-grade glioma" |
+      sample_id %in% hgg_lesions_df$sample_id,
+    experimental_strategy == "RNA-Seq",
+    RNA_library == "stranded"
+  )
+
+# Write to file
+readr::write_tsv(hgg_metadata_df, file.path(results_dir, "hgg_histologies.tsv"))
+
+#### Filter expression data ----------------------------------------------------
+
+# Filter to HGG samples only -- we can use hgg_metadata_df because it is subset
+# to RNA-seq samples
+stranded_expression <- stranded_expression %>%
+  dplyr::select(hgg_metadata_df$Kids_First_Biospecimen_ID)
+
+# Log2 transformation
+norm_expression <- log2(stranded_expression + 1)
+
+# Norm data mean and sd
+norm_expression_means <- rowMeans(norm_expression, na.rm = TRUE)
+norm_expression_sd <- apply(norm_expression, 1, sd, na.rm = TRUE)
+
+# Subtract mean
+expression_zscored <-
+  sweep(norm_expression, 1, norm_expression_means, FUN = "-")
+
+# Divide by SD remove NAs and Inf values from zscore for genes with 0 in normData
+expression_zscored <-
+  sweep(expression_zscored, 1, norm_expression_sd, FUN = "/") %>%
+  dplyr::na_if(Inf) %>%
+  na.omit()
+
+# Save matrix with all genes to file for downstream plotting
+readr::write_rds(expression_zscored,
+                 file.path(results_dir, "hgg_zscored_expression.RDS"))
+
+#### Filter focal CN data ------------------------------------------------------
+
+# Filter focal CN to ATRT samples only
+cn_metadata <- cn_df %>%
+  dplyr::left_join(select_metadata,
+                   by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::select(gene_symbol,
+                sample_id,
+                Kids_First_Participant_ID,
+                biospecimen_id,
+                status,
+                cytoband) %>%
+  dplyr::filter(sample_id %in% hgg_metadata_df$sample_id) %>%
+  dplyr::distinct()
```

Comment:
Can you add a comment to explain why you are using `distinct()`. I'm not very familiar with the contents of the `controlfreec_annotated_cn_autosomes.tsv.gz` so if you can give some insight to where the duplicates are coming from and whats the rationale of dropping them that'd be good.